### PR TITLE
Fix chunk loading deadlock and dangling chunk memory

### PR DESCRIPTION
**Pull request by kralverde**
If the chunk fetcher spawns and calls level.fetch_chunks and the chunk sender has not spawned yet, because the function is synchronous, the channel can be filled with no consumer

### CODE REVIEW

**Comment by Snowiiii** [comment]
After testing, It seems you don't unload the chunks somewhere resulting in huge memory consumption

**Comment by kralverde** [comment]
@Snowiiii could you take another look at the memory consumption?

**Comment by Snowiiii** [comment]
> @Snowiiii could you take another look at the memory consumption?

I will bit later, Not working on Pumpkin and just bit relaxing for the first time :D.

**Comment by Snowiiii** [comment]
Can confirm, memory is now consistent at 8-10MB again

**Comment by kralverde** [comment]
@StripedMonkey how do these changes look?

File: pumpkin-world/src/level.rs [hunk 1]
```diff
@@ -119,77 +127,96 @@ impl Level {
...
+    fn load_chunk_from_save(
+        chunk_reader: Arc<dyn ChunkReader>,
+        save_file: SaveFile,
+        chunk_pos: Vector2<i32>,
+    ) -> Result<Option<Arc<ChunkData>>, ChunkReadingError> {
+        match chunk_reader.read_chunk(&save_file, &chunk_pos) {
+            Ok(data) => Ok(Some(Arc::new(data))),
+            Err(
+                ChunkReadingError::ChunkNotExist
+                | ChunkReadingError::ParsingError(ChunkParsingError::ChunkNotGenerated),
```
Comment (StripedMonkey) [inline pumpkin-world/src/level.rs:154]: Why are these separate things?

**Comment by StripedMonkey** [comment]
> @StripedMonkey how do these changes look?

I don't have any objections to the PR code itself. It's looking pretty good.

**Comment by kralverde** [comment]
Streams chunks instead of collecting them in a vec and fixes a bug where chunks were not unloaded

**Pull request merged**